workflow-runtime/src/main/java/com/squareup/workflow1/SimpleLoggingWorkflowInterceptor.kt

```diff
@@ -111,4 +114,41 @@ public open class SimpleLoggingWorkflowInterceptor : WorkflowInterceptor {
   }
 
   protected open fun logError(text: String): Unit = System.err.println(text)
+
+  private fun formatLogMessage(
+    name: String,
+    session: WorkflowSession,
+    extras: Array<out Pair<String, Any?>>
+  ): String = if (extras.isEmpty()) {
+    "$name($session)"
+  } else {
+    "$name($session, ${extras.toMap()})"
+  }
+
+  private inner class SimpleLoggingContextInterceptor<P, S, O>(
+    private val session: WorkflowSession
+  ) : RenderContextInterceptor<P, S, O> {
+
+    override fun onActionSent(
+      action: WorkflowAction<P, S, O>,
+      proceed: (WorkflowAction<P, S, O>) -> Unit
+    ) {
+      logMethod("onActionSent", session, "action" to action) {
+        proceed(action)
+      }
+    }
+
+    override fun onRunningSideEffect(
+      key: String,
+      sideEffect: suspend () -> Unit,
+      proceed: (key: String, sideEffect: suspend () -> Unit) -> Unit
+    ) {
+      // Note that because logMethod is inline, it automatically acts like a suspend function here.
+      // The START will be logged immediately, and the END will be logged when the side effect
+      // completes, asynchronously.
```

Comment:
This comment is now out of date. And this whole method implementation is probably wrong. I think it still makes sense for it to log the lifetime of the actual side effect, as it was originally written.